### PR TITLE
Added pointer checking to sanity checks

### PR DESCRIPTION
**Pull request by TheDarkula**
r? @oli-obk

### CODE REVIEW

**Comment by TheDarkula** [comment]
cc @RalfJung

File: src/librustc_mir/interpret/validity.rs [hunk 1]
```diff
@@ -118,6 +108,24 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M> {
...
+                        return validation_failure!(
+                                "a pointer",
+                                path,
+                                format!("the type {}", ty.sty)
+                            );
+                    }
+                    ty::RawPtr(_) |
+                    ty::Ref(_, _, _) |
+                    ty::FnPtr(_) => {}
+                    _ => { unreachable!(); }
```
Comment (RalfJung) [inline src/librustc_mir/interpret/validity.rs]: Wouldn't it be much easier to handle this in `validate_scalar`? There's already a type match there to special-case `Char`; that could be used to do an *extra* check for the integer types.

Comment (TheDarkula) [reply src/librustc_mir/interpret/validity.rs]: I can move it into the main body if you like

Comment (oli-obk) [reply src/librustc_mir/interpret/validity.rs]: @RalfJung this is validate_scalar. The char match is when we know it's no pointer, so we can't do it there

Comment (RalfJung) [reply src/librustc_mir/interpret/validity.rs]: Oh. Not enough diff context. The never mind me.^^

**Comment by RalfJung** [comment]
You removed two ui tests, didn't they come with `.stderr` files that should also be removed?

**Comment by pietroalbini** [comment]
@craterbot run start=master#1114ab684fbad001c4e580326d8eb4d8c4e917d3 end=try#aa245a5862056b681d2ccdb178f58891108cd3a6 mode=check-only

**Comment by TheDarkula** [comment]
@RalfJung Those two tests were actually flagged with `//compile-pass `, so they had no '.stderr`. I checked :).
Thanks to both you and @pietroalbini for getting the crater test going

**Comment by pietroalbini** [comment]
This experiment is running quite slowly on crater, maybe it would be best to also do a perf run?

@rust-timer build aa245a5862056b681d2ccdb178f58891108cd3a6

**Comment by oli-obk** [comment]
Some noise in the timer, mostly even positive

Spurious regression in the crater run (thread '<unnamed>' panicked at 'procedural macro API is used outside of a procedural macro', libproc_macro/lib.rs:1295:13)

@bors r+

**Pull request merged**